### DISCUSSION TITLE
Would absolutely love an answer to #192.  Also  Gallery Downloads not working and Errors

### DISCUSSION THREAD

**Discussion by unspokentruths** [aclap-dev/video-downloadhelper#214] [Help and Q&A]
### Your question

Hey y'all.  So to make this as easy as possible i tried replicating this everywhere i could by searching online  for picture galleries so i could get hits with VDH to download galleries. and i always get the same errors when trying this on any url and hitting the analyze page button to generate hits of the picture files.
for example on this random url: https://pixabay.com/images/search/tiger/
after hitting analyze page, i proceed to press the arrow to  download the gallery, which should bring up the tab to choose where to save. instead-
the red bubble error quickly happens and when i click on it it says:

"e.core_media is undefined
moz-extension://1e1387e5-441d-4acf-bbf2-2d8511aaddbe/background/main.js:46
_T@moz-extension://1e1387e5-441d-4acf-bbf2-2d8511aaddbe/background/main.js:46:8743"

Never ever in years of using this before the last update did i get any errors for gallery downloads or gallery link downloads.
I really love the app and will definitely donate to it if these issues are resolved. 

Also happens in the brave browser with any url with any gallery download
using the above example on brave browser the following error is this:

 "Cannot read properties of undefined (reading 'container')
TypeError: Cannot read properties of undefined (reading 'container')
    at _T (chrome-extension://lmjnegcaeklhafolokijcfjliaokphfk/background/main.js:46:8758)
    at async Zr (chrome-extension://lmjnegcaeklhafolokijcfjliaokphfk/background/main.js:46:12889)"

### Addon & CoApp version and details

version: 8.2.1.1
channel: stable
build date: 2024-02-19
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-US
license: unset
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:123.0) Gecko/20100101 Firefox/123.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "C:\\Users\\banana-chan\\Downloads\\DownloadHelper\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\banana-chan",
  "converterBinary": "C:\\Users\\banana-chan\\Downloads\\DownloadHelper\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360955"
}

Brave Browser version and details:
version: 8.2.1.1
channel: stable
build date: 2024-02-19
build options: linuxlic: false, noyt: true, browser: chrome
lang: en-US
license: unset
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/122.0.0.0 Safari/537.36
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "C:\\Users\\banana-chan\\Downloads\\DownloadHelper\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\banana-chan",
  "converterBinary": "C:\\Users\\banana-chan\\Downloads\\DownloadHelper\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360955"
}

**Comment by loucapunheta** [comment]
I have the same error:

"e.core_media is undefined

moz-extension://f94e27e3-4c6b-498d-bf66-72a5e48d4606/background/main.js:46
_2@moz-extension://f94e27e3-4c6b-498d-bf66-72a5e48d4606/background/main.js:46:9018"

**Video DownloadHelper 8.2.1.3a1 (beta)**
Video DownloadHelper
Version 8.2.1.3a1 (beta)
Browser locale: en-US
Production build
Built on 2024-02-22
Build options: browser=firefox, linuxlic=false, noyt=false, target=mozilla

**Platform**
Platform Win x86-64
Browser Mozilla Firefox 123.0

**Companion App installed**
Found companion app: VdhCoApp 2.0.19
Companion app binary: C:\Program Files\DownloadHelper CoApp\vdhcoapp.exe

**version: 8.2.1.3a1**
channel: beta
build date: 2024-02-22
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-US
license: unset
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:123.0) Gecko/20100101 Firefox/123.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\humungu5",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360955"
}

**Comment by bchannell** [comment]
I have it, too.
"e.core_media is undefined"
I'll spare you my details.

**Comment by paulrouget** [comment]
I confirm the bug. I'll look into it asap!

Thanks.

**Reply by loucapunheta** [reply]
Hello,
Is not solved yet... dowload gallery does not ok!

**Reply by kwikdraw159** [reply]
Since the last update.  I can no longer download image galleries.  I get a "e.core_media is undefined error".  Sorry, I cannot assist with the coding explanations.  Your "asap" message was two months ago.  How about a fix, perhaps a VDH update?  I've been using and supporting VDH for years.  While I'm at it, the VDH creators should have a direct link, not this support page.  I want direct answers and solutions.

**Reply by loucapunheta** [reply]
Gallery download does not ok.
"e.core_media is undefined error" :(

```
version: 8.2.2.8a1
channel: beta
build date: 2024-03-14
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-US
license: unset
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:124.0) Gecko/20100101 Firefox/124.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\humungu5",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360955"
}

```

**Comment by ChrisGrabowy** [comment]
I use this extension with Firefox and still receive the same error message.

I have tried to use other extensions but I have not found one that works the same way as VDH.  

When VDH works then it seems to cleanly download the relevant picture files from a webpage.

Its April and no updates from the owner of the extension.

Any suggestions for another extension option?

**Comment by Mattmjs** [comment]
If VDH was used to download pictures like the tiger gallery above, did it get them as thumbnail images or full sized images ?
I'm asking since Firefox can do it but only as thumbnail images.

**Reply by ChrisGrabowy** [reply]
Not as thumbnails, but as full size picture files.

**Comment by sanosari** [comment]
I was suffering from the same problem. I had version 9 installed yesterday and the download option for picture galleries seems completely disappeared. Or at least, I cannot find it and do not see any comments on it. I would like to see it available again, and repaired in teh older versions of VDH. Please.

**Reply by ChrisGrabowy** [reply]
You can select the Use Legacy UI under More Settings.  And under the Legacy interface you can Analyze Page and find pictures.  But when downloading pictures it still reports the same error.

The fact the new version and UI do not provide such an option seems to imply that it will no longer download pictures.

**Comment by kwikdraw159** [comment]
Can the VDH people just fix the bug!!!  What's taking so long?

**Comment by paulrouget** [comment]
@kwikdraw159 @ChrisGrabowy @sanosari 

Sorry for the slow reply.

The download gallery system is indeed broken and has been for a while. Almost nobody uses this, so I kind of dropped the ball on that one.

Mind me asking: do you use Video DownloadHelper exclusively for the gallery feature, or it's just a thing on top of downloading videos? Just trying to understand the use case.

**Reply by ChrisGrabowy** [reply]
If by "gallery system", you mean the feature to download the relevant pictures from a webpage, then yes.  I use both equally.

**Comment by kwikdraw159** [comment]
Hello Paul,
     I'm a Premium user and purchased a license key years ago.  I use the
download gallery system (Links) all the time. It's one of the best features
of VDH. I also frequently use it for video downloads.  VDH is the only app
I could ever find that does this easily, quickly and efficiently.  I'd love
to see it restored.
Thank you for addressing this issue.
Mike C

**Reply by GrampaWildWilly** [reply]
Well, @paulrouget Paul, if you're going to restore the picture gallery download feature, might I piggyback a request to restore the functionality that permitted downloading zip files?  Yes, the browser can do zip downloads about as easily as VDH could.  The difference is that when you use the browser to download zip files -- any files, actually -- you get this stupid Window's file attribute that the file is "blocked."  I'm not sure what is not permitted on a file when it's blocked.  The warning in the file properties dialog says something to the effect of, "You downloaded this scary file off the Internet.  Goodness only knows what dragons are lurking in here who would love to burn you up & eat you.  But if you're some kind of brave hero, you can take off the blocked attribute.  But you will be to blame if anything bad happens."  Something like that.  I always figure it's safer to just take the attribute off.  I guess that makes me a hero.  When VDH used to download zip files, this stupidity didn't happen.

So.  VDH used to be able to download zip files.  Now it can't.  I would love for it to regain that ability.

**Comment by tmr75** [comment]
Will the developer please identify which versions (extension and coapp) we could roll back to where the "gallery system" still works?
I would like to use that until this is fixed, but don't want to play trial-and-error to find what works.
